lib/engine/game/g_18_fl.rb

```diff
@@ -111,6 +111,66 @@ def hotel_value
         @phase.status.include?('hotels_doubled') ? 20 : 10
       end
 
+      # How far is the start_hex from one of the corporation's stations?
+      # This is by track, not as the crow flies.
+      # start_hex is either a City or a Town Part. Cannot be Jacksonville
+      #
+      # NOTE: 18FL does not have forks in plain track, only junctions
+      # Also because of this, we only need to return the hexes visited
+      #  and can use the exits on tiles, except....
+      # The disjoint cities in Jacksonville are a problem though
+      def distance_to_station(corporation, start_hex)
+        raise GameError, 'Cannot calculate token cost of Jax' if start_hex.id == 'H23'
```

Comment:
make sure this method is not called when game is loading

raise GameError unless @game.loading

Reply:
I will have to add the calculated token cost to the Token action to make it work when the game is loading

Reply:
doens't the token action have a cost?

Reply:
No, It uses the price of the token from the corporation. In the case of 1861/1867, when it is time to place a token, the cost is easily calculable from the newly tokened hex and all of the pre-existing tokens (it's x distance as crows flies)

```
      def args_to_h
        {
          'city' => @city.id,
          'slot' => @slot,
          'tokener' => @tokener&.id,
          'token_type' => @token&.type == :normal ? nil : @token&.type,
        }
      end
```

Reply:
Is there anything preventing us from simply adding a mutable field to the Token action (cost) and modifying that in the step's logic?

Reply:
nope, although why would you have to mutate it? can't you just create it with the right price?

Reply:
I was able to put something together, let me know if that looks good